JavaScript development/Frameworks/React.md

```diff
@@ -0,0 +1,489 @@
+# React Guidelines and Best Practices
+
+*Guides are not rules and should not be followed blindly. Use your head and think.*
+
+## File Organization and Naming
+- as it is a good practice to divide React components as container and presentational components, you should place them in separate folders (example: **containers** and **components**)
+
+```
+...
+├─ components
+├─ containers 
+...
+```
+
+- use **camelCase** for general folder naming and **PascalCase** for React components, their stylesheet files and their containing folders 
+- each component type's root folder should have an **index.js** file which exports all components, so you can later import them from a single place
+
+```
+...
+├─ components
+⎮   ├─ LoginContent
+⎮   ⎮   └─ LoginContent.js
+⎮   ⎮   └─ LoginContent.scss
+⎮   ├─ LoginHeader
+⎮   ⎮   └─ LoginHeader.js
+⎮   ⎮   └─ LoginHeader.scss
+⎮   └─ index.js
+├─ containers 
+⎮   ├─ Login
+⎮   ⎮   └─ Login.js
+⎮   ⎮   └─ Login.scss 
+⎮   └─ index.js
+...
+```
+
+## Component Declaration
+- don't use React.createClass
+- use ES2015 **class syntax** for component declaration
+
+``` javascript
+// bad
+const ComponentName = React.createClass({
+  // ...
+  render () {
+    return (
+      <div>
+        {this.componentState.message}
+      </div>
+    );
+  }
+});
+    
+// good
+import React, {Component} from 'react';
```

Comment:
I would go with `React.Component` here. Scales well with TS, and will make sense soon.